src/csharp/Grpc.IntegrationTesting/ServicesGrpc.cs

```diff
@@ -47,12 +47,28 @@ public interface IBenchmarkServiceClient
     }
 
     // server-side interface
+    [System.Obsolete("Service implementations should inherit from the generated abstract base class instead.")]
     public interface IBenchmarkService
     {
       Task<global::Grpc.Testing.SimpleResponse> UnaryCall(global::Grpc.Testing.SimpleRequest request, ServerCallContext context);
       Task StreamingCall(IAsyncStreamReader<global::Grpc.Testing.SimpleRequest> requestStream, IServerStreamWriter<global::Grpc.Testing.SimpleResponse> responseStream, ServerCallContext context);
     }
 
+    // server-side abstract class
+    public abstract class BenchmarkServiceBase
```

Comment:
I think it might be clearer to generate this without the "Base" part, given that this is the level of inheritance that most clients will actually want to deal with. So just "BenchmarkService" in this case, and "Health" above. I think the "Impl" suffix for the concrete implementation is okay though.